tests/Common/ComponentsEventsTest.php

```diff
@@ -0,0 +1,533 @@
+    {
+        $config = $this->getConfiguration();
+        $this->components->addConfiguration($config);
+        // delete
+        $this->components->deleteConfiguration(self::COMPONENT_ID, $this->configurationId);
+        $events = $this->listEvents('storage.componentConfigurationDeleted');
+        $this->assertEvent(
+            $events[0],
+            'storage.componentConfigurationDeleted',
+            'Deleted component configuration "test" (wr-db)',
+            $this->configurationId,
+            'test',
+            'componentConfiguration',
+            [
+                'component' => 'wr-db',
+                'configurationId' => $this->configurationId,
+                'name' => 'test',
+                'version' => 2,
+            ]
+        );
+
+        // purge
+        $this->components->deleteConfiguration(self::COMPONENT_ID, $this->configurationId);
+        $events = $this->listEvents('storage.componentConfigurationPurged');
+        $this->assertEvent(
+            $events[0],
+            'storage.componentConfigurationPurged',
+            'Permanently deleted component configuration "test" (wr-db)',
+            $this->configurationId,
+            'test',
+            'componentConfiguration',
+            [
+                'component' => 'wr-db',
+                'configurationId' => $this->configurationId,
+                'name' => 'test',
+                'version' => 2,
+            ]
+        );
+    }
+
+    public function testConfigurationRestore()
+    {
+        $config = $this->getConfiguration();
+        $this->components->addConfiguration($config);
+        $this->components->deleteConfiguration(self::COMPONENT_ID, $this->configurationId);
+
+        $this->components->restoreComponentConfiguration(self::COMPONENT_ID, $this->configurationId);
+        $events = $this->listEvents('storage.componentConfigurationRestored');
+
+        $this->assertEvent(
+            $events[0],
+            'storage.componentConfigurationRestored',
+            'Restored component configuration "test" (wr-db)',
+            $this->configurationId,
+            'test',
+            'componentConfiguration',
+            [
+                'component' => 'wr-db',
+                'configurationId' => $this->configurationId,
+                'name' => 'test',
+                'version' => 3,
+            ]
+        );
+    }
+
+    public function testConfigurationVersionCopy()
+    {
+        $config = $this->getConfiguration();
+        $this->components->addConfiguration($config);
+
+        $newConfig = $this->components->createConfigurationFromVersion(
+            self::COMPONENT_ID,
+            $this->configurationId,
+            1,
+            'new'
+        );
+
+        $events = $this->listEvents('storage.componentConfigurationCopied');
+        $this->assertEvent(
+            $events[0],
+            'storage.componentConfigurationCopied',
+            'Created component configuration "new" (wr-db)',
```

Comment:
Přijde mi divná ta message - mělo by být `copied configuration 'old' to 'new' (wr-db)`

Reply:
Trochu se mi nelíbí, že budeme (a potenciálně i zákazníci) muset myslet na to, že "vytvoření konfigurace" se může objevit jako několik různých event - created, restored, copied... Nevím, jestli by nebylo lepší těch event vyhodit několik - na všechny ty věci. Tj. created+restored, created+copied. Takhle to má různé formáty těch parametrů a tak... ale to specifikoval @ErikZigo , takže to je možná dotaz spíš na něj. Rozhodně to nemusíme tímhle blokovat, ale zvážil bych jestli to neupravit.

Reply:
> Přijde mi divná ta message - mělo by být copied configuration 'old' to 'new' (wr-db)
@zajca muzes to pls upravit? Tak jak rika Tomas?

> Trochu se mi nelíbí, že budeme (a potenciálně i zákazníci) muset myslet na to, že "vytvoření konfigurace" se může objevit jako několik různých event 
Tohle bych fakt teď neřešil

Reply:
> Tohle bych fakt teď neřešil

OK, ale jakmile to bude venku a budou to používat zákazníci, tak to bude epická migrace a ještě dlouho se toho nezbavíme.

Reply:
A jak to budou používat zákaznící? A jak používají současné Storage Api eventy?

Reply:
Más nějaký přiklad kde se to teď takhle na straně Storage API event dělá, že by se pro jednu akci posílalo několik event?

Například přijmutí pozvánky do projektu se zaloguje taky jen jako přijmutí pozvánky.
Podle tvého modelu by se tam teda mělo logovat zároveň i přidání uživatele do projektu.

Reply:
Ano. Protože to je reálně ta část, co tě zajímá. Jestli to bylo přidáním adminem, pozvánkou, nebo že to uživatel vyhrál v loterii. Mě zajímá, že mi přibyl nový uživatel v projektu.

Reply:
Ale zákazníka možná ne. Chtěl jsem prostě upozornit na to, že to možná není domyšlené proti tomu use-casu.

Reply:
Use-casů může být milion a tady to nevyřešíš. Nejlepší když hodíš do backlogu issue.

Reply:
Text sem fixnul.